File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int n = prices.length;
        int buyPrice = Integer.MAX_VALUE;
        for(int i =0;i<n;i++){
            buyPrice = Math.min(buyPrice,prices[i]);
            maxProfit = Math.max(maxProfit,prices[i] - buyPrice);
        }
        return maxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.c
int min(int a,int b){
    return a > b ? b : a;
}
int max(int a,int b){
    return a > b ? a: b;
}
int maxProfit(int* prices, int pricesSize) {
    int maxProfit = 0;
    int minPrice = prices[0];
    for(int i= 0;i<pricesSize;i++){
        minPrice = min(minPrice,prices[i]);
        maxProfit = max(maxProfit,prices[i] - minPrice);
    }
    return maxProfit;
}